src/test/java/test/dependent/github1380/GitHub1380Sample4.java

```diff
@@ -0,0 +1,21 @@
+package test.dependent.github1380;
```

Comment:
The behavior the same since the hierarchy of order features was introduced.

TestNG respect the most important feature feature first (dependsOnMethods here).

We must support circular dependency when many features because of the others priority features like group-by-instance which may produce circular dependencies.
Feel free to block it and tests, you'll see many errors.